### ISSUE TITLE
Add visually prominent focus styles for all focusable elements

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#6886]
### Description
There doesn't seem to be a standard focus style for focusable elements.

It looks like anchors/links use the browser default focus style (some type of outline), and buttons have removed the browser default focus style for a more subtle one (slight background colour change).

For example, in this image, the left button is in focus and the right one is not:
![Slightly dark gray button next to slightly lighter gray button.](https://user-images.githubusercontent.com/17884198/93941710-dfc18980-fce3-11ea-9f98-82601274e45b.png)

Related: #7289

Some areas have focusable elements with no distinguishable focus style at all. For example, the sidebar navigation links:
![Sidebar navigation link with current link styled differently, but no distinguishable style on the focused link.](https://user-images.githubusercontent.com/17884198/93942544-79d60180-fce5-11ea-8104-de18e3d15d08.png)

Also the profile dropdown in the top corner (which is also getting cut off a bit by my scrollbar):
![Profile icon button.](https://user-images.githubusercontent.com/17884198/93942674-be619d00-fce5-11ea-8117-da8dcf3de87c.png)

I think this issue has 2 parts:
1. Don't completely remove outline from focus styles, even if there is a custom focus style. [If we make the outline colour transparent instead of removing the outline entirely, the outline will be preserved and recoloured for Windows High Contrast Mode](https://sarahmhigley.com/writing/whcm-quick-tips/#1.-custom-focus-styles-%2B-a-transparent-outline).

2. The custom focus styles should be more visually prominent. Having a more consistent style across all focusable elements (e.g. anchors/links, buttons, and inputs) would make it easier to keep track of where focus is. 

### Steps to reproduce

1. Use keyboard to tab through interactable elements such as anchors/links, buttons, and inputs.

### Additional info

- Craft version: 3.5.11.1

**Comment by brandonkelly** [comment]
Merged #7680 for the next release! 🎉

**Comment by brandonkelly** [comment]
Craft 3.6.12 is out now with this ✨
